.github/workflows/master.yml

```diff
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
+
+name: Master workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+
+      - name: Build
+        run: go build
+
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build and push Docker image
```

Comment:
This only `builds` the Docker image, but it does not `push` it. Should we also add `docker-push`? Should `docker-push` be applicable only for `push` to master? Any thoughts on this? Moreover, are you planning to do this same thing for all other repos (including the UPF)?

Reply:
The current CI does not `docker push` anywhere and this change here keeps this behaviour. We may add it in the future but this was not the purpose of this change.

And yes the goal is to improve consistency between workflows of the different NF's. The workflow may end up being identical for all of them but it can be much closer than what it is right now.

Reply:
I changed the description to only contain "Build Docker image"